be/src/service/internal_service.cpp

```diff
@@ -1163,14 +1163,14 @@ void PInternalServiceImplBase<T>::list_fail_point(google::protobuf::RpcControlle
 }
 
 template <typename T>
-Status PInternalServiceImplBase<T>::_exec_short_circuit(brpc::Controller* cntl, const PExecShortCircuitRequest*,
+Status PInternalServiceImplBase<T>::_exec_short_circuit(brpc::Controller* cntl, const PExecShortCircuitRequest* request,
                                                         PExecShortCircuitResult* response) {
     auto ser_request = cntl->request_attachment().to_string();
     std::shared_ptr<TExecShortCircuitParams> t_requests = std::make_shared<TExecShortCircuitParams>();
     {
         const auto* buf = (const uint8_t*)ser_request.data();
         uint32_t len = ser_request.size();
-        RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, TProtocolType::BINARY, t_requests.get()));
+        RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, request->attachment_protocol(), t_requests.get()));
```

Comment:
I have a question before the input is TProtocolType, but how do you convert it as a string input.

Reply:
And you need to consider back compatable issues.
When the new version BE work with old FE, there is no attachment_protocol() in the request, this will cause faults.
You should avoid these back-compatible faults.

Reply:
there is a function with the same name to accept `string` type of the protocol.

no compatable issues here, as now BE will ignore optional `attachment_protocol`, guaranteed by thrift.